lib/SyTest/ApplicationService.pm

```diff
@@ -0,0 +1,91 @@
+package SyTest::ApplicationService;
+
+use strict;
+use warnings;
+
+use Future::Utils qw( repeat );
+
+sub new
+{
+   my $class = shift;
+   my ( $info, $await_http ) = @_;
+
+   my $path = $info->path;
+
+   # Map event types to ARRAYs of Futures
+   my %futures_by_type;
+
+   my $f = repeat {
+      $await_http->( qr{^\Q$path\E/transactions/\d+$}, sub { 1 },
+         timeout => 0,
```

Comment:
The same as the clearer `undef`; now changed to that plus added comment. 65e7b66